### ISSUE TITLE
Maps handling in repo

### ISSUE THREAD

**Issue by Indigo744** [neveldo/jQuery-Mapael#341]
@neveldo 

I was wondering how to better integrate maps with Mapael.

Right now, some maps are defined in this repo (in `/js/maps`) and some others mainly in `mapael-maps` repository.

The best approach would be first to store everything to the `mapael-maps` repository and no maps in this repo. The examples file could simply use the rawgit link to access them.

Also, what about using [submodules](https://github.com/blog/2104-working-with-submodules)? We could line the folder `/js/maps` to `mapael-maps`...

What do you think?

**Comment by neveldo** [comment]
Hello @Indigo744 ,

I agree with you on the point that having some maps in jQuery-mapael repository and some other in mapael-maps repository is not a good idea. As you said, we could merge all the maps within mapael-maps repository, and make the example load the maps from rawgit.

I've never used git submodules, what could be our benefits of using it to include maps within jquery-mapael repository ? Does it works well is NPM / bower (ie : does the user will get all the maps if he requires jquery-mapael ?). That said, maybe I'm wrong, but I think it can be misleading as most of users aren't familiar with git submodules...

Having all the maps bundled into one repository ease the users to contribute to it by adding new maps. 

But having one repository per map bring some other advantages. Each map will live their life (versionning, npm package releases) independantly from the others. It would allow users to load through NPM only the maps they need, and to benefits from newer versions through NPM if a bugfix has been provided for instance. But, the main drawback I think is that it will be a lot more complicated for users to contribute by adding their own maps to a repository. They will need to create a new repository for the map, NPM config, maintain it, etc. After all, i'm not sure one repo per map is the way to go ...

**Comment by Indigo744** [comment]
Yeah, maybe submodules are weird and would introduce more question than answers :)

Regarding maps repo, this is difficult. On one hand, user with NPM would love to refer to only a specific repo for a specific map, on the other hand, it would be a maintenance nightmare... 

Anyway, nobody did complain about it, so I guess it's not such a bad approach.

**Comment by neveldo** [comment]
I agree with you on this point. Regarding the move of the maps from jquery-mapael to mapael-maps , I think it should be planified for the next major release, at it will introduce a non backward compatible change !
